src/lib/operators/join_helper/join_output_writing.cpp

```diff
@@ -0,0 +1,308 @@
+#include "join_output_writing.hpp"
+
+#include <unordered_map>
+
+#include <boost/functional/hash_fwd.hpp>
+
+#include "hyrise.hpp"
+#include "scheduler/job_task.hpp"
+#include "storage/segment_iterate.hpp"
+
+namespace {
+
+using namespace hyrise;  // NOLINT(build/namespaces)
+
+using PosLists = std::vector<std::shared_ptr<const AbstractPosList>>;
+using PosListsByColumn = std::vector<std::shared_ptr<PosLists>>;
+
+struct PosListsHasher {
+  size_t operator()(const PosLists& pos_lists) const {
+    return boost::hash_range(pos_lists.begin(), pos_lists.end());
+  }
+};
+
+/**
+ * Returns a vector where each entry with index i references a PosLists object. If entries i and j reference the same
+ * PosLists object, both columns originate from the same table. As a consequence, they can later share the same (newly
+ * created) PosLists of the join result.
+ * @param input_table
```

Comment:
use `@return` and `@param` properly?

Reply:
Better?

`* @param input_table The input reference table for which the pos list mapping is built.`

Reply:
Did not want to force you to write doxygen style stuff. Just lookwd werid that `Return` and `@param input_table` were kond of mixed and the `@param` description was empty. You can as well just remove enything but the textual description

Reply:
Ahh, just checked the two doxygen lines and not the entire comment. Moved the comment about the returned vector.

```
/**
 * @param input_table the input reference table for which the PosList mapping is built
 * @return Returns a vector where each entry with index i (for column i of input_table) references a PosLists object.
 *         If entries i and j reference the same PosLists object, both columns originate from the same table. As a
 *         consequence, they can later share the same (newly created) PosLists of the join result.
 */
```

We have a quite a lot of doxygen comments. Never understood why, but I don't want to start changing them. :-/

Reply:
Hm, I guess it's nice to get what a public method does quickly if you might apply it without actually reading the code. Quite clear what happens and what to pass. Does not make sense for a helper method inside a cpp AND the anonymous namespace though. Don't care as long it does not look too weird (which was the case IMO)